### PR TITLE
TESTS enable numpy.core tests run

### PR DESCRIPTION
**Pull request by densmirn**
1359 passed, 3043 skipped, 9 xfailed, 1 warning in 18.23s

### CODE REVIEW

File: dpnp/dparray.pyx [hunk 1]
```diff
@@ -48,7 +48,7 @@ cimport dpnp.dpnp_utils as utils
 
 
 # initially copied from original
-cdef class _flagsobj:
+class _flagsobj:
```
Comment (samir-nasibli) [inline dpnp/dparray.pyx:51]: Do we really need changes related with dparray here? wouldn't it be better to move it to a separate PR, that could be merged before this one.

File: dpnp/dpnp_iface_trigonometric.py [hunk 2]
```diff
@@ -663,14 +662,13 @@ def log(x1):
     [0.0, 1.0, 2.0, -inf]
 
     """
+    if use_origin_backend(x1):
+        if not isinstance(x1, dparray):
+            pass
```
Comment (samir-nasibli) [inline dpnp/dpnp_iface_trigonometric.py:667]: It is better to move this fixes into other PR. Or provide, please, full description for this PR.

File: dpnp/dparray.pyx [hunk 3]
```diff
@@ -531,6 +549,14 @@ cdef class dparray:
         if order is not b'C':
             utils.checker_throw_value_error("dparray::reshape", "order", order, b'C')
 
+        if dn:
```
Comment (samir-nasibli) [inline dpnp/dparray.pyx:552]: This additions apply serious changes on our API and may lead big difference from the original `numpy`. Please, move this changes into other PR and add request review also from O. Pavlyk.

**Comment by shssf** [comment]
@densmirn It is looks what we need. Need to continue this work.
Also, I would propose to change subdirectory name from `tests/third_party/numpy` to `tests/third_party/numpy_ext` to have a chance in future to distinguish tests used from external package internally (like cupy now) and direct use of external test system.

It is also possible (optional) to move such test runner outside of DPNP tests to something like `dpnp/tests_external`

**Comment by densmirn** [comment]
> @densmirn It is looks what we need. Need to continue this work.
> Also, I would propose to change subdirectory name from `tests/third_party/numpy` to `tests/third_party/numpy_ext` to have a chance in future to distinguish tests used from external package internally (like cupy now) and direct use of external test system.
> 
> It is also possible (optional) to move such test runner outside of DPNP tests to something like `dpnp/tests_external/numpy`

@shssf how do you think it makes sense to move functionality changes to separate PRs or not? Anyway all the tests are passed.

**Comment by shssf** [comment]
@densmirn I think all things are ok but let me sometime to review it. No need to change this PR at this time.

**Review by shssf — APPROVED** [review]

**Comment by samir-nasibli** [comment]
@shssf @densmirn I vote to split this PR into logically smaller ones and request review for some of them from @oleksandr-pavlyk
